### ISSUE TITLE
 Video unavailable

### ISSUE THREAD

**Issue by simmstein** [omarroth/invidious#811]
Hello, 

**Versions**

* Invidious version:  0.19.1-e03b4b7 (sha256 of the compiled file 271291b36749ee7bae51c3834d08da8eb8bc7e4842525489c56a3399ef0b5d17)
* OS: Debian 10 (amd64)
* Crystal version: 0.31.1 [0e2e1d067] (2019-09-30)
* Database: Postgresql 9.6+181+deb9u2
* Logs: empty (no error)

**What happened**

I click on a video's thumbnail and the message "Video unavailable." is shown.

**Expected**

Play the video clicked.

**Information**

The video is played using invidio.us (0.19.1-e03b4b7 master).

**Comment by simmstein** [comment]
From the server:

```
$ youtube-dl "https://www.youtube.com/watch?v=XXXXXXXXX"
[youtube] -OZ--RQcaY4: Downloading webpage
[youtube] -OZ--RQcaY4: Downloading video info webpage
WARNING: unable to download video info webpage: HTTP Error 429: Too Many Requests
WARNING: unable to download video info webpage: HTTP Error 429: Too Many Requests
WARNING: unable to download video info webpage: HTTP Error 429: Too Many Requests
WARNING: unable to download video info webpage: HTTP Error 429: Too Many Requests
WARNING: unable to download video info webpage: HTTP Error 429: Too Many Requests
Traceback (most recent call last):
  File "/usr/bin/youtube-dl", line 11, in <module>
    load_entry_point('youtube-dl==2019.1.17', 'console_scripts', 'youtube-dl')()
  File "/usr/lib/python3/dist-packages/youtube_dl/__init__.py", line 477, in main
    _real_main(argv)
  File "/usr/lib/python3/dist-packages/youtube_dl/__init__.py", line 467, in _real_main
    retcode = ydl.download(all_urls)
  File "/usr/lib/python3/dist-packages/youtube_dl/YoutubeDL.py", line 2002, in download
    url, force_generic_extractor=self.params.get('force_generic_extractor', False))
  File "/usr/lib/python3/dist-packages/youtube_dl/YoutubeDL.py", line 793, in extract_info
    ie_result = ie.extract(url)
  File "/usr/lib/python3/dist-packages/youtube_dl/extractor/common.py", line 508, in extract
    ie_result = self._real_extract(url)
  File "/usr/lib/python3/dist-packages/youtube_dl/extractor/youtube.py", line 1672, in _real_extract
    token = video_info.get('token') or video_info.get('account_playback_token')
AttributeError: 'NoneType' object has no attribute 'get'
```

To be confirmed: this is not a issue from invidious.

**Comment by atahanacar** [comment]
12 hours ago, I had the same issue, but the issue is no longer there right now. Seems like Youtube changed some policies regarding connections.

**Comment by Perflyst** [comment]
Your IP got blocked by Google, other instances face the same issues.

An example to test if you are blocked:
```
root@invidious:~# curl 'https://www.youtube.com/watch?v=ukDht4jRcT0&gl=US&hl=en&disable_polymer=1&has_verified=1&bpctr=9999999999'
<HTML><HEAD><meta http-equiv="content-type" content="text/html;charset=utf-8">
<TITLE>302 Moved</TITLE></HEAD><BODY>
<H1>302 Moved</H1>
The document has moved
<A HREF="https://www.google.com/sorry/index?continue=https://www.youtube.com/watch%3Fv%3DukDht4jRcT0%26gl%3DUS%26hl%3Den%26disable_polymer%3D1%26has_verified%3D1%26bpctr%3D9999999999&amp;hl=en&amp;q=EgRf2BjmGJjv9uwFIhkA8aeDS2A2A_pZaQC7pukXkNR0Cy6xim-gMgFy">here</A>.
</BODY></HTML>
```

**Comment by PureTryOut** [comment]
I have the same error since like a week, but I get a full result if I try that test of @Perflyst.

**Comment by simmstein** [comment]
By using my server as a proxy, I had to fill recaptcha to play a video on youtube.com (private tab).

**Comment by OdinGitDat** [comment]
I have the same for every video on v0.19.1-d36c536 vanilla docker container.

**Comment by omarroth** [comment]
You will also want to check the output for `/das_captcha` which will also indicate rate-limiting:

```
$ curl 'https://www.youtube.com/watch?v=ukDht4jRcT0&gl=US&hl=en&disable_polymer=1&has_verified=1&bpctr=9999999999' -s | grep '/das_captcha
<form action="/das_captcha" method="POST">
```

Testing with youtube-dl should return a `429` as well if this is the case.

As @Perflyst mentioned, this is due to rate-limiting/IP banning by Google. This has been an issue for other projects as well, see fent/node-ytdl-core#444, Tyrrrz/YoutubeExplode#292, and ytdl-org/youtube-dl#21729.

If IPv6 is available for your box, I would recommend testing with `force_resolve: ipv4` or `force_resolve: ipv6` in your config. Otherwise, you'll unfortunately have to wait for Google to unban you or get a new IP.

**Comment by OdinGitDat** [comment]
When I use a browser on my server to watch a video on youtube it asks for a captcha and then lets me watch videos for as long as the cookie stays in the browser.

Isn't there a way to relay this captcha to invidious webmasters and use the cookie for subsequent requests?

**Comment by jaruiz59** [comment]
Yep getting the same message .Video unavailable.

**Comment by nchristensen** [comment]
You can always host it locally to avoid the IP banning. On Ubuntu (and Debian?) install docker-compose and docker.io from the repositories, clone this repository, and then call docker-compose up inside the root of the repository. It will take a while to build but will be available at localhost:3000 when it finishes.

Hosting it locally will not shield your IP from YouTube, but you can always use a VPN with it.

**Comment by asorel1942** [comment]
I don't quite understand what everyone in this thread means by hosting invidio locally. Is there a way to utilize the service without a browser? If so, will this be able to circumvent the "video is unavailable" messages everyone seems to be getting?

**Comment by BurnyBoi** [comment]
Does this mean invidio.us has been IP blocked by Google?

**Comment by nchristensen** [comment]
> Is there a way to utilize the service without a browser?

You can run the Invidious server as a program on your computer and access your very own functional Invidious instance at http://localhost:3000 using a browser on that computer. It will only be available on that computer, but this is probably sufficient for many people. (There are ways to make it available to anyone on your home network or even to anyone on the internet, however.)

> If so, will this be able to circumvent the "video is unavailable" messages everyone seems to be getting?
It should as long as you are accessing videos at a normal rate. 

> Does this mean invidio.us has been IP blocked by Google?
As I understand Google is just limiting the rate of requests. Functionally, the result is essentially the same.

**Comment by HumanG33k** [comment]
Doing that they provide bad experience and people want to move back to YT. It’s a better option that blocking IP which provide badbuzz. They have done some similar things for ie when they downgrade qualities based on user agent in order to push FF users to Chrome.

**Comment by OdinGitDat** [comment]
> You can always host it locally to avoid the IP banning.
@nchristensen Can you please stop spreading this as a solution? The OP specifically has this problem on his own instance and so do I. Yes I am the only user of my instance, I watch max 10 videos a day, and I've still been banned twice within 3 days. Both my bans lasted 12 hours and afaik they all do.

I don't like to be pessimistic, but I think this may be the end of invidious if relaying a captcha/cookie is impractical as omarroth said.

**Comment by coolcoderplus** [comment]
Why not redirect instance traffic to tor? The ip of the instance will be protected from being banned.

**Comment by simmstein** [comment]
@coolcoderplus it could work but this is not a fix.

**Comment by Gert-dev** [comment]
@coolcoderplus I regularly use Tor and am often blocked by Google services due to "too much traffic". The list of endpoints is constantly changing, but likely the majority remains fairly constant. As also only a subset of Tor nodes function as actual endpoints, the pool of IP addresses is also said subset.

In a grossly simplifying sense, it's comparable to a dynamic list of proxies, of which the IP addresses end up flagged after a while.

**Comment by sentriz** [comment]
i am experiencing this on my home network too. youtube-dl doesn't work either. but surprisingly newpipe on my phone is working

**Comment by 2secslater** [comment]
I have a somewhat public instance (other folk can access it) running on my home network, and has been for rougly three days now. I have had the same IP address, using the instance most of the time and I haven't been sniped by Google (yet).

**Comment by m4teh** [comment]
I've also been hit by this (self hosting Invidious via outbound VPN provider).
They must have done a big VPN sweep recently. I get a CAPTCHA when I browse YouTube's website too.

Is work being put into adding an ability to perform the CAPTCHA from the Invidious interface? That's the easiest solution I can think of.

**Comment by inhji** [comment]
My solution was to snapshot the whole instance and restore it to a different VPS with a new ip address.

**Comment by HumanG33k** [comment]
it s hard workaround not really easly setup for people with less money/skills.

I guess there is no way to contact google for help us … they probably say use the api and we loose some privacy improvement provide by invidious.

**Comment by atahanacar** [comment]
@HumanG33k This is kind of unrelated but I am curious. If I am using an off-site server as a proxy, does it matter if I am using the API or an extractor?

**Comment by HumanG33k** [comment]
@atahanacar hope it can help :

if i m not wrong yes you can use an off-site server as a proxy. 
For me there is differences between an extractor and an api call if it s not open(need an auth).
You also need to know if you are alone or not.

if i understand well the server make request to google insteed of you.

First case (standard usage) :
you (auth or not) --> google
login or not google know who you are.

Case 2 :
you && server --> google
google know 1 or more people comes from here (on same ip) its quite easy to know what you see if you are alone or other people don’t use the server but if you don’t use some tools (addon/ads blocker) google can know who you are.

Case 3 :
you --> server --> google 
It s a little more difficult for google to know who you are but if he knows you use the server and nobody else then he knows.

Case 4 :
you --> server --> random --> google
this can be the better but also the hard to maintain case. to be less track (the main goal of google) you should never be the you google knows. the random should also be a several random, for me the most usefull random should be multiple random, between 1 to N by channel or topic. all random should also be used by several people (vpn, tor, for ie) 

For the api vs scrapping the main issue with the api is you need to be auth in most of the case and 1 auth = 1 account. And google can know how many people exist in the worst case. (how many people look videos at the same time for ie) but api are created in order to provide data not fancy (useless) stuff like ui.

Crawler in other way use the (browser) human way to grab content but because fancy things are provide the content of a page can change and it can make tools which download hard to maintain. 

For me the good way should be 

you --> the One app --> server : with p2p and federation, and can mirror(grab and store) content (peertube or mediagobelin) --> random --> multiple providers (vimeo, YT, dailymotion, p2p, etc)

**Comment by Pentaphon** [comment]
Many videos are showing this for me now.

**Comment by BurnyBoi** [comment]
Started my own instance a week ago that only I use, and now this is already happening to me. Not an IP ban since I can use mpv to watch YT videos just fine. This is crazy. Google seems to have some way of knowing that a video is being requested by an Invidious server. Can we hide that?

**Comment by HumanG33k** [comment]
i think about that last days. 

i don’t know how @omarroth send request to YT but maybe we have to improve the fake network request. And fake at if we are Firefox or other but maybe not on a chromium/chrome base. I’m pretty sure they have some proprietary request come from chrome (with token, etc). they probably also have some other UI things to be prevent scraping. I also see a survey on a not loggin youtube page but i have to leave and i don’t do it. I think they maybe try again to kill FF. That can interesting to set up an MITM to check network packets from/to youtube.

**Comment by 2secslater** [comment]
I don't know if this is worth noting or not, but my DNS server is getting a constant stream of queries for www.youtube.com, every second.




24 Oct 2019, 11:03 by notifications@github.com:

**Comment by atahanacar** [comment]
> What would that change? YouTube also blocks you if there are too many requests from a normal IP with a normal browser (see VPN or Tor network)

I have a private instance that only I and a few friends use. It is blocked, but not by IP, as I can use my OpenVPN to watch YouTube videos successfully. The only reason I can't use Invidious is because of this silly block. I didn't even make that many requests, 10 videos a day at max.

**Comment by HumanG33k** [comment]
I can not use my invidious instances but youtubes with the same exit ip address sometimes. 
i will make test to see that.

I also suspect they have an ai for filter incoming request. If it can learn we are legitimate trafic it can be good. 

For how google manage all things https://www.bloomberg.com/news/articles/2019-10-23/google-accused-of-creating-spy-tool-to-squelch-worker-dissent

**Comment by elypter** [comment]
maybe invidio.us should consider mirroring more metadata to minimize the request count. the distributed effort to rescue the annotations was quite successful.

**Comment by jee-r** [comment]
we'd need to put P2P in there.

**Comment by elypter** [comment]
there already are distributed videoplatforms like bitchute. no need to reinvent the wheel. its probably hard enough to integrate them into the project and the biggest problem is that as soon as you no longer just proxy content but host it you have to manage it and are legally responsible. no matter if its invidio.us who is responsible or the individual p2p users doesnt matter.

if it's just the metadata then thats a lot easier. and just the metadata can work without p2p as well

**Comment by defterade** [comment]
Started getting this on all videos today. I don't appear to be IP banned (youtube-dl, mpv, curl showed no issues).

Running latest flourgaz/invidious.

**Comment by BurnyBoi** [comment]
As an update to mine above, seems @m4teh is on the right track here. I noticed if I go to youtube.com directly on my server and go to watch a video, I get a recaptcha before the video will play. Obviously the Invidious instance is currently unable to solve this recaptcha. if Invidious could embed the recaptcha on the page, allow the user to solve it, and pass the result back to YouTube, this seems like it would do the trick. Also, it seems like once you store some cookie saying you solved it, it doesn't ask again, so maybe Invidious could do that too?

**Comment by NettoHikari** [comment]
I'm also hit by a ban since today. I'm hosting invidious since 2 days for me, my wife and 2 or 3 friends. It is publicly available, though. youtube-dl also doesn't work on the VPS the Invidious instance is running on. Very sad.

**Comment by cy8aer** [comment]
The captcha problem looks like this really old one in searx: https://github.com/asciimoo/searx/issues/729

**Comment by cy8aer** [comment]
probably useful? https://github.com/unixfox/antibot-proxy

**Comment by ncaos** [comment]
Magically works again for me today, after a few weeks. Temp ban removed, maybe?

**Comment by defterade** [comment]
> Magically works again for me today, after a few weeks. Temp ban removed, maybe?

For me too.

**Comment by m4teh** [comment]
Did you guys update Invidious? I saw some recent commits for captcha/user agent which may have a positive effect?

I’ll have to switch my VPN tunnel back to check if mine is still blocked or not.

**Comment by ncaos** [comment]
No update.

**Comment by defterade** [comment]
I didn't update either.

**Comment by cy8aer** [comment]
Related?

https://www.bleepingcomputer.com/news/google/chrome-78-disables-code-integrity-check-to-mitigate-aw-snap-crashes/

**Comment by 2secslater** [comment]
No. That's an issue with the Chrome browser on client-ends.

**Comment by cy8aer** [comment]
Sorry I was too slow with deleting the post but because @2secslater answered I put it in again (but off topic)

Related?

https://www.bleepingcomputer.com/news/google/chrome-78-disables-code-integrity-check-to-mitigate-aw-snap-crashes/

**Comment by simmstein** [comment]
Without updating invidious, I can watch videos from my instance…For how much longer?

**Comment by BurnyBoi** [comment]
`force_resolve: ipv4` Seemed to fix it for me...for now. So I guess they banned/limited my ipv6 address, but not ipv4.

**Comment by Perflyst** [comment]
A lot of instances started working again (v4 and v6). Either Google was testing some new "bot detection" or they had a bug. Nobody knows. Please do not spam this issue that yours is working again, thanks. For further discussion you can also join #invidious on freenode or #invidious:matrix.org

**Comment by BurnyBoi** [comment]
A single message isn't spam lol. Close it then

**Comment by atahanacar** [comment]
A single message isn't spam, but 10 messages with the same content is.

**Comment by 2secslater** [comment]
Can we please stay on-topic?

**Comment by BurnyBoi** [comment]
Sure...I never sent 10 messages with the same content though :)

**Comment by sentriz** [comment]
my home instance started working again 👍
I also recently deleted my browser's sessions/history, but maybe that's not the reason it's fixed

**Comment by atahanacar** [comment]
I think that was a one time Google ban that will probably no longer happen after commits like e05a25d701a2b4765028d9f23306098b263e15b9. This issue should be closed in my opinion.

**Comment by RoiArthurB** [comment]
@atahanacar I don't think so, I have my personal instance and the official Invidious one are out one more time (like here : https://invidio.us/watch?v=2VSFnIm8TMQ ) :/

**Comment by ncaos** [comment]
My personal instance is down again and it hasn't been used in days, as far as I know. Your link above works for me @RioArthurB.

**Comment by RoiArthurB** [comment]
@ncaos The official invidious link was down (not anymore), but my personal is still down :/

**Comment by gerroon** [comment]
I am still blocked, I think Google started blocking users who use non standard apps randomly. I get errors with regular youtube-dl as well.

**Comment by Pentaphon** [comment]
>I get errors with regular youtube-dl as well.

I don't. I only get errors with invidious.

**Comment by simoniz0r** [comment]
invidio.us and a couple of other instances seem to have gotten blocked again within the last half hour or so.  It seems like Google is really starting to crack down on this.

**Comment by Vistaus** [comment]
I have the same issue on the official Invidious site. Also, when I try to log in using my Google account, Invdious reports that my account is temporarily disabled. But my account works fine on the official YouTube site.

**Comment by omarroth** [comment]
Should be fixed with d46b26e3bc87fbbdfa9550820410041b861bb393.

**Comment by aidanharris** [comment]
Do we know what triggers the blocking? Google claims it's too much traffic and I'm willing to give them the benefit of the doubt. As another user whose had their home network blocked I want to assume there's no malice in Google's actions here and do whatever it is I can do in order to appease them to prevent further blocks from occurring.

As a gesture of good will I've temporarily blocked all YouTube domains via DNS and will see if that helps. I'm the only user of my instance so it's hard to imagine I'm sending them so much traffic. My IP is semi-static too (it's technically dynamic but it doesn't really change often at all) so not something that's shared with lots of users like a VPN or proxy server.

**Comment by simoniz0r** [comment]
AFAIK, Google blocks IPs that request their pages too frequently. Swapping out the User-Agent may help, but I kind a feel like the only way to actually stop this is to setup rate limiting that prevents Invidious from making too many requests to Google.

**Comment by blixnood** [comment]
Thank Google for getting blocked. GoogleBot was killing my server (okay, it was running fine but adding the usual load would start to cause problems) by scraping every single video. I blocked it and now suddenly my server is fine. There was no reason for my instance to get blocked (same applies to most/all of you) yet it was happening over and over again. Now *I* know why, and I'm sure it's the reason that it has happened to most people, too.

**Comment by elypter** [comment]
i assume it's simply because google can see the difference between requests from invidio.us and youtube.com. they dont like that. to get around that you have to immitate original requests. rate limiting might not be enough and comes with compromises.

**Comment by m4teh** [comment]
It’s quite simple really. YouTube doesn’t want you using their stuff via third party software. They want you using their software via their website, so they can collect all metrics and what not. 

This isn’t an overload issue. 
YouTube page still works fine (often without captcha) generally when they ban my instance.

**Comment by aidanharris** [comment]
@m4teh The captcha's are real (if you really are being blocked/rate-limited). For reasons I don't quite understand YouTube can still fingerprint you in private browsing mode though so they could just trust your shadow profile so not show you a captcha. if I can figure out why this happens this seems like a bug in firefox (doesn't happen with Chromium's private browsing mode it shows me a captcha)? A clean profile with `firejail --private --no-dbus -- /usr/bin/firefox https://youtube.com` shows me a captcha.

**Comment by simoniz0r** [comment]
This seems to be happening more frequently.  My guess would be that as Invidious is getting more popular, it's making more requests to Youtube to scrape data, which leads to it getting blocked more frequently as it's going over the limit of requests allowed in a certain amount of time to Google websites.
As I said before, I really think the only way this is going to be solved is by rate limiting Invidious to make sure it does not go above the amount of requests that Google allows from one IP in a certain amount of time.

**Comment by narcolepticinsomniac** [comment]
Damn. I didn't realize this had been going on for a while. Had no issues up till today, but now it's practically unusable.

**Comment by Vagmer** [comment]
> This seems to be happening more frequently. My guess would be that as Invidious is getting more popular, it's making more requests to Youtube to scrape data, which leads to it getting blocked more frequently as it's going over the limit of requests allowed in a certain amount of time to Google websites.

That concerns an individual Invidious instance becoming more popular. Though, doesn't the instance need to scrape video metadata pre-emptively anyway before even the first user connects, and after first scraping is done theoretically most users (not enabling proxy or Dash) don't even increase the number of bandwidth pulled from Google.
The other side of this is not directly related to instance popularity: Google isn't limited to keeping their ratelimiting, banning and potential 3rd-party-software-detection algorithms static over time. This is clearly happening more frequently (i.e. often) now as compared to a couple of months ago because they have tightened said algorithms on their side, and they can keep changing them and experimenting with them over time.
This has always been the vulnerability of Invidious, it has an obvious single point of failure. I fear for the future of Invidious in its current working model, I don't know what can really be done in practical terms. Interested in hearing what @omarroth has to say and if there are any possible upcoming plans or tricks. At least if current blocking presents a captcha to the instance, it appears that there are known working software solutions for bypassing/solving Recaptcha today, so this may be a method to provide additional leeway, at least until Google plugs up that out, too.

> As I said before, I really think the only way this is going to be solved is by rate limiting Invidious to make sure it does not go above the amount of requests that Google allows from one IP in a certain amount of time.

That's the very obvious "solution", though one that might mean in effect killing or at the very least crippling the functioning of the instance a lot. Unless you keep making all the requests as before, but split them through different proxies/shards with different IPs so individual IPs don't cross said limit. So that means either finding or paying for a lot of servers (let's go ahead and imagine a successful crowdfunding campaign, even) which can still end of failing at the end (for example due to blacklists and/or possible legal action against servers), or shifting to a P2P model ala BitTorrent / PeerTube.

**Comment by Gert-dev** [comment]
The notion of PeerTube is an interesting one. One can imagine a community around PeerTube instances that download and seed most videos that are now on YouTube itself, effectively rendering YouTube itself unnecessary in the long term, as long as at least one user keeps seeding said videos. It would also take the load of a central server and make it much harder to take down videos.

Pondering aside and IANAL, but it wouldn't surprise me if that were also illegal.

**Comment by Vagmer** [comment]
Indeed. At the moment, it's not like the most widespread uses of P2P tech tend to be completely legal... Even Invidious instances get some DMCA notices sent to them already, even though they're just providing a proxy and a web client to YouTube and not actually storing YouTube content (long-term, anyway). Though of course IANAL but I wouldn't be surprised if Invidious can already be said to be breaking some law or other right now as it is (though naturally applicability of laws may actually "depend on how good your lawyer is"). Certainly, it's not just against Google's interests but also doesn't follow the YouTube or Google ToS and causes financial losses to them.

**Comment by elypter** [comment]
The easiest short term solution could be to add a "Watch on <other invidious instance>" link below "Watch on Youtube". it may seem kinda pointless because the page has already been loaded on invidio.us but it avoids any traffic created by people clicking on related videos and it makes people much more likely to consider switching to another instance permantently.

**Comment by Pentaphon** [comment]
> The easiest short term solution could be to add a "Watch on " link below "Watch on Youtube".
Or encourage people to play videos with an addon + youtube-dl and just make invidious a UI.

**Comment by Vagmer** [comment]
To tell the truth I'm not entirely sure what the last two comments exactly mean. Seems to be some context missing. 

elypter: looks like you're trying to say that official main instance (with presumably highest number of hits?) **invidio.us** should some more effort into making all users aware that other instances exist and that they might be faster for them too. That's a good idea in general. In fact might as well bake this feature into all instances by default and instance admin can turn it off if he wants. I made a new issue for this concept: #944.

Pentaphon: what? Invidious is already pretty much an alternative UI with proxy capabilities for browsing and watching. Are you suggesting removing the proxy capabilities? There have already been various alternative YouTube clients/viewers for different platforms for accessing YT directly before (and after) Invidious was made. Doing that for Invidious doesn't seem very useful at all, if even feasible.

Apparently disabling watching-proxy on the instance doesn't even always help (please share info if you have it) as instances doing this still face issues I think, and allegedly some self-hosted / private instances are suffering, too. These suggest meta-data scraping alone without active [proxy / any] users watching can be enough to trigger current YouTube blocking/algorithms.

**Comment by elypter** [comment]
> To tell the truth I'm not entirely sure what the last two comments exactly mean. Seems to be some context missing.

changed "Watch on " to "Watch on <other invidious instance>". had to escape bracket characters

**Comment by simoniz0r** [comment]
It would seem as if either most instances cannot afford to keep buying new IPs or Google is banning them *much* more quickly.  There's only 2 working https instances as far as I can tell right now.

**Comment by sentriz** [comment]
does anyone know how NewPipe are doing it? it's been working for me still for ~2 months, same home IP as invidious

**Comment by tleydxdy** [comment]
newpipe uses your IP to access youtube (afaik), but invidious uses the server IP

**Comment by sentriz** [comment]
I meant to say I'm selfhosting invidious so newpipe and invidious would be accessing youtube from my same home IP I think

**Comment by elypter** [comment]
maybe ipv6 helps if you get a sizable subnet for your server. google can still block a whole range but they might be more hesitant, especially if they dont know the network topology

**Comment by gerroon** [comment]
I think that Google is using Webrtc and Browser stuff to discover different computers inside the same network, the reason I am saying is that I am blocked by Youtube but not my wife's computer access to Youtube is not blocked and we are on the same wifi.

**Comment by sentriz** [comment]
hmm @elypter using a static ipv4 address at home

**Comment by haizrul** [comment]
Problem happen again. Most instances are down. :(

**Comment by tleydxdy** [comment]
it's not suprising that they can detect devices using the same IP, google basically live off of that.

**Comment by narcolepticinsomniac** [comment]
I only got blocked for a day. I assume [this commit](https://github.com/omarroth/invidious/commit/a017574f74665edc9645b9e1c0414e0a34e13f41) helped, because of the timing. Still having connectivity issues though. 

I only use invidious as a fallback with a script which redirects on error. The majority of the time, if youtube's blocked an embed for whatever reason, a redirect to invidious pulls the vid. Ever since it started kinda working again, at least half the time, I'm getting `waiting for invidio.us...`.

**Comment by elypter** [comment]
>They might be tagging the "good" ones and letting them have access and block everything else from that network.

this is all in theory preventable. its just a matter of how much effort you put in to mimic a normal browser. the problem is that solutions that work one day might not work anymore the next. so decentralization should always be of similar importance

**Comment by Vagmer** [comment]
NewPipe is quite different from Invidious, even if self hosted. NewPipe is just an alternative YT player/browser. It's like using a different, specialized web browser to access `youtube.com`. Or like using youtube-dl to do so.

**Comment by simoniz0r** [comment]
I've noticed that [vid.wxzm.sx](https://vid.wxzm.sx) seems to be the most reliable instance for getting a list of available videos even though it is blocked from getting data for individual videos.

This has got me thinking a bit... what if you were to add an option to Invidious that would handle most of the calls to YouTube on the client side through JavaScript?  You could still allow people to use the method of getting the info from the invidious server if they wish, but running the YouTube calls client side through JavaScript would allow an instance to keep working even if it has been blocked.

**Comment by elypter** [comment]
sounds good as a last resort but it should only be applied if distributing the work and caching is not enough.

**Comment by Vagmer** [comment]
Yeah... as I discussed in a previous comment https://github.com/omarroth/invidious/issues/811#issuecomment-560462889 that would only be good as a last ditch effort to just keep the site(s) working in name only, basically, if forsaking most of their original purpose. At that point there's little difference from viewing the site/running the JavaScript from a local HTML file instead.

**Comment by elypter** [comment]
there is still a great difference. there is no code made by google running on your system. and the amount of requests to google servers is still vastly reduced.

**Comment by Perflyst** [comment]
> but running the YouTube calls client side through JavaScript would allow an instance to keep working even if it has been blocked

-> "Does not require JS to play videos" in the readme

I am strongly against this feature, use youtube.com if your invidious instance is blocked or tell the admin to use anti-captcha.com (which currently solves all problems)

**Comment by simoniz0r** [comment]
> -> "Does not require JS to play videos" in the readme
> 
> I am strongly against this feature, use youtube.com if your invidious instance is blocked or tell the admin to use anti-captcha.com (which currently solves all problems)

This would be an ***option***, not a ***requirement***.  You could still use Invidious without JavaScript with what I suggested above.

**Comment by Vagmer** [comment]
Kind of tired of repeating the same thing - maybe it's totally useless to, and I never said anything new to begin with  - but again, that's basically asking Invidious to become, or add an option to be, a locally running YouTube client (viewer), which is almost a completely different kind of project than it is now, one which has little benefit of being hosted on a remote website at all, and doesn't even have to be running within a web browser. It would actually be a waste of any Invidious development time, unless that becomes the only way for the project to kind-of-live, I guess. If this ("connecting to and parsing YouTube in an alternative way") is what you want then you don't strictly need Invidious. Many alternative YT clients already exist for various platforms and have existed before Invidious, the uniqueness of which is that it's not really a client.

**Comment by Vagmer** [comment]
@Perflyst 
> tell the admin to use anti-captcha.com (which currently solves all problems)

Good to hear. You know of instances currently employing such a solution? This one seems really really cheap, too. I'm sure I and others wouldn't mind donating to help make use of such.

**Comment by omarroth** [comment]
> You know of instances currently employing such a solution?

The two that I can recall are https://invidio.us and https://invidious.snopyta.com, although I suspect others are also using this feature.

Closing since this has been addressed by d46b26e and 8236036. I would recommend taking a look at [this](https://github.com/omarroth/invidious/issues/957#issuecomment-576424042) comment in #957 for more explanation.

**Comment by Perflyst** [comment]
it is https://invidious.snopyta.org, not .com ;)

Jan 20, 2020 22:33:50 Omar Roth <notifications@github.com>:

**Comment by TROMsite** [comment]
I am too wondering how to try and fix this with our instance https://ytb.trom.tf/ - we have ipv6 enabled already. We keep the instance updated and all that. So basically what happens is youtube blocking the server's IP? That's the main issue? If so, is there any fix in sight for this?

**Comment by syeopite** [comment]
> I am too wondering how to try and fix this with our instance https://ytb.trom.tf/ - we have ipv6 enabled already. We keep the instance updated and all that. So basically what happens is youtube blocking the server's IP? That's the main issue? If so, is there any fix in sight for this?

Yes. It's been implemented for quite awhile now.  [Please checkout the documentation page on how to setup anti-captcha](https://github.com/iv-org/documentation/blob/master/How-to-setup-Anti-Captcha.md).

And for the record, please don't comment on such an ancient thread!

**Comment by TROMsite** [comment]
Ah I see. So you pay for an anticaptcha to get around this... Shouldn't some proxy solutions be better? Maybe I'll open another issue so we don't revive such zombie threads. Thanks!

**Comment by unixfox** [comment]
> Ah I see. So you pay for an anticaptcha to get around this... Shouldn't some proxy solutions be better? Maybe I'll open another issue so we don't revive such zombie threads. Thanks!

Proxy support is tracked in https://github.com/iv-org/invidious/issues/1036.

**Comment by syeopite** [comment]
Alternatively #879 could be a possible solution. Either way, I'm locking this issue.
